File: js/tests/littleDictionary.test.js
const dictionary = require("../littleDictionary");

const data = [
  ["bu", ["button", "breakfast", "border"], ["button"]],
  ["beau", ["pastry", "delicious", "name", "boring"], []],
  ["there", ["there", "here", "everywhere"], ["there"]],
  ["sno", ["snow", "paper", "ghost"], ["snow"]],
  [
    "ma",
    ["marriage", "married", "many", "delight"],
    ["marriage", "married", "many"]
  ],
  [
    "no",
    ["inferno", "notion", "nominate", "note", "fairy"],
    ["notion", "nominate", "note"]
  ],
  [
    "cre",
    ["creating", "creature", "creed", "increasing"],
    ["creating", "creature", "creed"]
  ]
];

describe("dictionary", () => {
  describe("Return words that start with initial word", () => {
    test.each(data)("dictionary(%p, %p) = %p", (initial, words, result) => {
      let actual = dictionary(initial, words);

      expect(actual).toEqual(result);
    });
  });
});

File: CSharp/Tests/LittleDictionaryTest.cs
using Xunit;

namespace CSharp.Tests
{
    public class LittleDictionaryTest
    {
        [Theory]
        [InlineData("bu", new string[] { "button", "breakfast", "border" }, new string[] { "button" })]
        [InlineData("beau", new string[] { "pastry", "delicious", "name", "boring" }, new string[] { })]
        [InlineData("there", new string[] { "there", "here", "everywhere" }, new string[] { "there" })]
        [InlineData("sno", new string[] { "snow", "paper", "ghost" }, new string[] { "snow" })]
        [InlineData("ma", new string[] { "marriage", "married", "many", "delight" }, new string[] { "marriage", "married", "many" })]
        [InlineData("no", new string[] { "inferno", "notion", "nominate", "note", "fairy" }, new string[] { "notion", "nominate", "note" })]
        [InlineData("cre", new string[] { "creating", "creature", "creed", "increasing" }, new string[] { "creating", "creature", "creed" })]
        public void Dictionary_StringInputAndStringArray_ReturnStringsThatStartWithInitialWord(string initial, string[] words, string[] expected)
        {
            var actual = LittleDictionary.Dictionary(initial, words);

            Assert.Equal(expected, actual);
        }
    }
}